uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestCalendarFeedDataTest.php

```diff
@@ -0,0 +1,518 @@
+    $absenceType = AbsenceTypeFabricator::fabricate(['title' => 'Sample Type', 'hide_label' => 1]);
+    $contact1 = ContactFabricator::fabricate(['first_name' => 'Contact1', 'last_name' => 'LastContact1']);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact1['id']],
+      ['period_start_date' => CRM_Utils_Date::processDate('today')]
+    );
+
+    $fromDate = new DateTime('+1 day');
+    $toDate = new DateTime('+2 days');
+
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contact1['id'],
+      'type_id' => $absenceType->id,
+      'from_date' => $fromDate->format('Y-m-d H:i:s'),
+      'to_date' => $toDate->format('Y-m-d H:i:s'),
+      'from_date_type' => '',
+      'to_date_type' => ''
+    ]);
+
+    //feed config is for all contacts in any department/location
+    $feedConfig1 = LeaveCalendarFeedConfigFabricator::fabricate([
+      'title' => 'Feed 1',
+      'composed_of' => [
+        'leave_type' => [$absenceType->id],
+      ]
+    ]);
+
+    $displayName = "{$contact1['first_name']} {$contact1['last_name']}";
+    $expectedResult[0] = [
+      'id' => $leaveRequest->id,
+      'contact_id' => $leaveRequest->contact_id,
+      //Leave label will be displayed for Admin user
+      'display_name' => $displayName . ' (Sample Type)',
+      'from_date' => $fromDate->format('Y-m-d H:i:s'),
+      'to_date' => $toDate->format('Y-m-d H:i:s')
+    ];
+
+    $leaveFeedData = new LeaveRequestCalendarFeedData($feedConfig1->hash);
+    $this->assertEquals($expectedResult, $leaveFeedData->get());
+  }
+
+  public function testGetWillNotReturnDisabledLeaveTypesData() {
+    $absenceType = AbsenceTypeFabricator::fabricate();
+    $absenceType2 = AbsenceTypeFabricator::fabricate();
+    $contact1 = ContactFabricator::fabricate(['first_name' => 'Contact1', 'last_name' => 'LastContact1']);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact1['id']],
+      ['period_start_date' => CRM_Utils_Date::processDate('today')]
+    );
+
+    $params[1] = [
+      'contact_id' => $contact1['id'],
+      'first_name' => $contact1['first_name'],
+      'last_name' => $contact1['last_name'],
+      'type_id' => $absenceType->id,
+      'from_date' => CRM_Utils_Date::processDate('+1 day'),
+      'to_date' => CRM_Utils_Date::processDate('+2 days'),
+    ];
+
+    //This leave request will not be returned because  the absence type is disabled
+    //just after the leave was created for it.
+    $params[2] = [
+      'contact_id' => $contact1['id'],
+      'first_name' => $contact1['first_name'],
+      'last_name' => $contact1['last_name'],
+      'type_id' => $absenceType2->id,
+      'from_date' => CRM_Utils_Date::processDate('today'),
+      'to_date' => CRM_Utils_Date::processDate('+2 days'),
+    ];
+
+    //disable absence type 2
+    AbsenceTypeFabricator::fabricate(['id' => $absenceType2->id, 'is_active' => 0]);
```

Comment:
Why don't you create it as disabled? Do we really need this separate statement to make it disabled?